featuretools/primitives/standard/transform_primitive.py

```diff
@@ -584,49 +587,75 @@ def email_address_to_domain(emails):
         return email_address_to_domain
 
 
-class NumericLag(TransformPrimitive):
+class Lag(TransformPrimitive):
     """Shifts an array of values by a specified number of periods.
 
     Args:
         periods (int): The number of periods by which to shift the input.
             Default is 1. Periods correspond to rows.
 
-        fill_value (int, float, optional): The value to use to fill in
+        fill_value (int, float, str, bool, optional): The value to use to fill in
             the gaps left after shifting the input. Default is None.
 
     Examples:
-        >>> lag = NumericLag()
-        >>> lag(pd.Series(pd.date_range(start="2020-01-01", periods=5, freq='D')), [1, 2, 3, 4, 5]).tolist()
+        >>> lag = Lag()
+        >>> lag([1, 2, 3, 4, 5], pd.Series(pd.date_range(start="2020-01-01", periods=5, freq='D'))).tolist()
         [nan, 1.0, 2.0, 3.0, 4.0]
 
         You can specify the number of periods to shift the values
 
-        >>> lag_periods = NumericLag(periods=3)
-        >>> lag_periods(pd.Series(pd.date_range(start="2020-01-01", periods=5, freq='D')), [1, 2, 3, 4, 5]).tolist()
-        [nan, nan, nan, 1.0, 2.0]
+        >>> lag_periods = Lag(periods=3)
+        >>> lag_periods(["hello", "world", "test", "foo", "bar"], pd.Series(pd.date_range(start="2020-01-01", periods=5, freq='D'))).tolist()
+        [nan, nan, nan, 'hello', 'world']
 
         You can specify the fill value to use
 
-        >>> lag_fill_value = NumericLag(fill_value=100)
-        >>> lag_fill_value(pd.Series(pd.date_range(start="2020-01-01", periods=4, freq='D')), [1, 2, 3, 4]).tolist()
+        >>> lag_fill_value = Lag(fill_value=100)
+        >>> lag_fill_value([1, 2, 3, 4], pd.Series(pd.date_range(start="2020-01-01", periods=4, freq='D'))).tolist()
         [100, 1, 2, 3]
     """
 
-    name = "numeric_lag"
+    name = "lag"
     input_types = [
-        ColumnSchema(semantic_tags={"time_index"}),
-        ColumnSchema(semantic_tags={"numeric"}),
+        [
+            ColumnSchema(logical_type=Double),
+            ColumnSchema(logical_type=Datetime, semantic_tags={"time_index"}),
+        ],
+        [
+            ColumnSchema(semantic_tags={"category"}),
+            ColumnSchema(semantic_tags={"time_index"}),
+        ],
+        [
+            ColumnSchema(logical_type=Boolean),
+            ColumnSchema(semantic_tags={"time_index"}),
+        ],
+        [
+            ColumnSchema(logical_type=BooleanNullable),
+            ColumnSchema(semantic_tags={"time_index"}),
+        ],
+        [
+            ColumnSchema(logical_type=AgeNullable),
```

Comment:
Do we want `AgeFractional` as well? (I imagine `Age` wasn't doable bc it used the non nullable `int64` dtype?)